src/emailmessage.cpp

```diff
@@ -17,6 +17,25 @@
 #include "qmailnamespace.h"
 #include <qmaildisconnected.h>
 #include <QTextDocument>
+#include <QTemporaryFile>
+#include <QStandardPaths>
+#include <QDir>
+
+// Part finder
+struct PartFinder {
+    PartFinder(QMailMessageContentType type, const QMailMessagePart *&part) : contentType(type), partFound(part) {}
```

Comment:
Humm, still would need to copy this property to another pointer inside the class in getCalendarPart, otherwise it's destructed when this goes out of scope